internal/querynodev2/segments/segment_loader.go

```diff
@@ -586,6 +592,18 @@ func (loader *segmentLoader) loadFieldsIndex(ctx context.Context, segment *Local
 		)
 
 		segment.AddIndex(fieldID, fieldInfo)
+
+		// set average row data size of variable field
+		field, err := schemaHelper.GetFieldFromID(fieldID)
+		if err != nil {
+			return err
+		}
+		if typeutil.IsVariableDataType(field.GetDataType()) {
+			err = segment.UpdateFieldRawDataSize(numRows, fieldInfo.FieldBinlog)
```

Comment:
OK, I got it. When the field was indexed, we use binlog size to set the average size.